substrate/client/rpc-spec-v2/src/chain_head/error.rs

```diff
@@ -22,52 +22,62 @@ use jsonrpsee::{
+	/// The follow subscription was started with `withRuntime` set to `false`.
+	#[error("The `chainHead_follow` subscription was started with `withRuntime` set to `false`")]
+	InvalidRuntimeCall(String),
+	/// Wait-for-continue event not generated.
+	#[error("Wait for continue event was not generated for the subscription")]
+	InvalidContinue,
 	/// Invalid parameter provided to the RPC method.
 	#[error("Invalid parameter: {0}")]
 	InvalidParam(String),
-	/// Invalid subscription ID provided by the RPC server.
-	#[error("Invalid subscription ID")]
-	InvalidSubscriptionID,
+	/// Internal error.
+	#[error("Internal error: {0}")]
+	InternalError(String),
+}
+
+/// Errors for `chainHead` RPC module, as defined in
+/// https://github.com/paritytech/json-rpc-interface-spec.
+pub mod rpc_spec_v2 {
+	/// The provided block hash is invalid.
+	pub const INVALID_BLOCK_ERROR: i32 = -32801;
+	/// The follow subscription was started with `withRuntime` set to `false`.
+	pub const INVALID_RUNTIME_CALL: i32 = -32802;
 	/// Wait-for-continue event not generated.
-	#[error("Wait for continue event was not generated for the subscription")]
-	InvalidContinue,
+	pub const INVALID_CONTINUE: i32 = -32803;
 }
 
-// Base code for all `chainHead` errors.
-const BASE_ERROR: i32 = 2000;
-/// The provided block hash is invalid.
-const INVALID_BLOCK_ERROR: i32 = BASE_ERROR + 1;
-/// Fetch block header error.
-const FETCH_BLOCK_HEADER_ERROR: i32 = BASE_ERROR + 2;
-/// Invalid parameter error.
-const INVALID_PARAM_ERROR: i32 = BASE_ERROR + 3;
-/// Invalid subscription ID.
-const INVALID_SUB_ID: i32 = BASE_ERROR + 4;
-/// Wait-for-continue event not generated.
-const INVALID_CONTINUE: i32 = BASE_ERROR + 5;
+/// General purpose errors, as defined in
+/// https://www.jsonrpc.org/specification#error_object.
+pub mod json_rpc_spec {
+	/// Invalid parameter error.
+	pub const INVALID_PARAM_ERROR: i32 = -32602;
+	/// Internal error.
+	pub const INTERNAL_ERROR: i32 = -32603;
+}
 
 impl From<Error> for ErrorObject<'static> {
 	fn from(e: Error) -> Self {
 		let msg = e.to_string();
 
 		match e {
-			Error::InvalidBlock => ErrorObject::owned(INVALID_BLOCK_ERROR, msg, None::<()>),
-			Error::FetchBlockHeader(_) =>
-				ErrorObject::owned(FETCH_BLOCK_HEADER_ERROR, msg, None::<()>),
-			Error::InvalidParam(_) => ErrorObject::owned(INVALID_PARAM_ERROR, msg, None::<()>),
-			Error::InvalidSubscriptionID => ErrorObject::owned(INVALID_SUB_ID, msg, None::<()>),
-			Error::InvalidContinue => ErrorObject::owned(INVALID_CONTINUE, msg, None::<()>),
+			Error::InvalidBlock =>
+				ErrorObject::owned(rpc_spec_v2::INVALID_BLOCK_ERROR, msg, None::<()>),
+			Error::InvalidRuntimeCall(_) =>
+				ErrorObject::owned(rpc_spec_v2::INVALID_RUNTIME_CALL, msg, None::<()>),
+			Error::InvalidContinue =>
+				ErrorObject::owned(rpc_spec_v2::INVALID_CONTINUE, msg, None::<()>),
+			Error::InvalidParam(_) =>
+				ErrorObject::owned(json_rpc_spec::INVALID_PARAM_ERROR, msg, None::<()>),
+			Error::InternalError(_) =>
```

Comment:
Technically according https://www.jsonrpc.org/specification#error_object the message should be "Internal error" but I guess this is fine anyway.